python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -1064,6 +1064,28 @@ def collect_set(col: "ColumnOrName") -> Column:
 collect_set.__doc__ = pysparkfuncs.collect_set.__doc__
 
 
+def listagg(col: "ColumnOrName", delimiter: Optional[Union[Column, str, bytes]] = None) -> Column:
+    if delimiter is None:
+        return _invoke_function_over_columns("listagg", col)
+    else:
+        return _invoke_function_over_columns("listagg", col, lit(delimiter))
+
+
+listagg.__doc__ = pysparkfuncs.listagg.__doc__
+
+
+def listagg_distinct(
+    col: "ColumnOrName", delimiter: Optional[Union[Column, str, bytes]] = None
+) -> Column:
+    if delimiter is None:
+        return _invoke_function_over_columns("listagg_distinct", col)
```

Comment:
I suspect this won't work, since we don't have `listagg_distinct` in `FunctionRegistry`.

In spark connect, `_invoke_function_over_columns` just build a unresolved function.

Reply:
I think you can refer to https://github.com/apache/spark/blob/05750def2da6fe16fc0cf8a3eee79bb2056df979/python/pyspark/sql/connect/functions/builtin.py#L1090-L1096